doc/manual/faq.rst

```diff
@@ -779,6 +779,59 @@ which is quite different. It is the empty (or "bottom") type, a type with no val
 and no subtypes (except itself). This is now written as ``Union()`` (an empty union
 type). You will generally not need to use this type.
 
+Memory
+------
+
+Why does ``x += y`` allocate memory when ``x`` and ``y`` are arrays?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In julia, ``x += y`` gets replaced during parsing by ``x = x + y``.
```

Comment:
Do people outside computer science understand what "syntax sugar" means? (That said, "during parsing" may not be much clearer.)